hamilflow/models/brownian_motion.py

```diff
@@ -0,0 +1,197 @@
+    """
+
+    x0: Union[float, int, List[Union[float, int]]] = 1.0
+
+    @field_validator("x0")
+    @classmethod
+    def check_x0_types(cls, v: Union[float, int, list]) -> np.ndarray:
+        if not isinstance(v, (float, int, list)):
+            raise ValueError(f"Value of x0 should be int/float/list of int/float: {v=}")
+
+        return np.asarray(v)
+
+
+class BrownianMotion:
+    r"""Brownian motion describes motion of small particles
+    with stochastic forces applied to them.
+    The math of Brownian motion can be modeled
+    with Wiener process.
+
+    For consistency, we always use
+    $\mathbf x$ for displacement, and
+    $t$ for steps. The model we are using is
+
+    $$
+    \begin{align}
+    \mathbf x(t + \mathrm dt) &= \mathrm x(t) +
+    \mathcal{N}(\mu=0, \sigma=\sigma \sqrt{\mathrm d t})
+    \end{align}
+    $$
+
+    References:
+
+    1. Brownian motion and random walks. [cited 13 Mar 2024].
+        Available: https://web.mit.edu/8.334/www/grades/projects/projects17/OscarMickelin/brownian.html
+    2. Contributors to Wikimedia projects. Brownian motion.
+        In: Wikipedia [Internet]. 22 Jan 2024 [cited 13 Mar 2024].
+        Available: https://en.wikipedia.org/wiki/Brownian_motion
+
+
+    !!! example "1D Brownian Motion"
+
+        The dimsion of our Brownian motion is specified by
+        the dimension of the initial condition.
+
+        To simulate a 1D Browian motion, we define the system and initial condition:
+
+        ```python
+        system = {
+            "sigma": 1,
+            "delta_t": 1,
+        }
+
+        initial_condition = {
+            "x0": 0
+        }
+        ```
+
+        The Brownian motion can be simulated using
+
+        ```python
+        bm = BrownianMotion(system=system, initial_condition=initial_condition)
+
+        bm(n_steps=100)
+        ```
+
+    !!! example "2D Brownian Motion"
+
+        To simulate a 2D Browian motion,
+
+        ```python
+        system = {
+            "sigma": 1,
+            "delta_t": 1,
+        }
+
+        initial_condition = {
+            "x0": [0, 0]
+        }
+
+        bm = BrownianMotion(system=system, initial_condition=initial_condition)
+
+        bm(n_steps=100)
+        ```
+
+    :param system: the Brownian motion system definition
+    :param initial_condition: the initial condition for the simulation
+    """
+
+    def __init__(
+        self,
+        system: Dict[str, float],
+        initial_condition: Optional[Dict[str, float]] = {},
+    ):
+        self.system = BrownianMotionSystem.model_validate(system)
+        self.initial_condition = BrownianMotionIC.model_validate(initial_condition)
+
+    @property
+    def dim(self) -> int:
+        """Dimension of the Brownian motion"""
+        return self.initial_condition.x0.size
+
+    @property
+    def _axis_names(self) -> List[str]:
+        return [f"y_{i}" for i in range(self.dim)]
+
+    def _trajectory(self, n_new_steps: float, seed: float) -> np.ndarray:
```

Comment:
- Why private `_`?
- Why is `seed` `float`, not `int`? [doc](https://numpy.org/doc/stable/reference/random/legacy.html#numpy.random.RandomState)

Reply:
1. `_` is internal use not private. Here we use it because it is intended for internal use in `__call__`.
2. Fixed in 8c799b1